src/config.rs

```diff
@@ -86,17 +87,20 @@ impl Config {
             metadata_cmd.manifest_path(path);
         }
         let metadata = metadata_cmd.exec().expect("get cargo metadata");
+        let manifest = metadata[&metadata.workspace_members[0]]
+            .manifest_path
+            .clone();
+        let lock_file = manifest.with_extension("lock");
```

Comment:
Could we base this off of the workspace root instead of assuming based on the location of the `Cargo.toml`?

https://github.com/rust-lang/cargo/blob/master/src/cargo/ops/lockfile.rs#L12 

Looks like that's how cargo decides if there is a `Cargo.lock` or not.